core/src/cluster_info.rs

```diff
@@ -653,6 +663,117 @@ impl ClusterInfo {
         *self.entrypoints.write().unwrap() = entrypoints;
     }
 
+    pub fn save_contact_info(&self) {
+        let nodes = {
+            let gossip = self.gossip.read().unwrap();
+            let entrypoint_gossip_addrs = self
+                .entrypoints
+                .read()
+                .unwrap()
+                .iter()
+                .map(|contact_info| contact_info.gossip)
+                .collect::<HashSet<_>>();
+
+            gossip
+                .crds
+                .get_nodes()
+                .filter_map(|v| {
+                    if let Some(contact_info) = v.value.contact_info() {
```

Comment:
maybe just 
```
let contact_info = v.value.contact_info().unwrap()
```
here.

Reply:
Ok.  Generally unwraps like this make me a little nervous, we've had crashes due to bad unwraps before.  But I guess in this case `get_nodes_contact_info()` already calls `get_nodes()` and explicitly panic on `None` so an unwrap here isn't any worse

Reply:
Agree about avoiding unwrap in general, but in this case it is an invariant that *has to be always the case*.